test/integration/crossPartition.spec.ts

```diff
@@ -867,6 +724,103 @@ describe("Cross Partition", function() {
       }
     });
 
+    it("Validate simple LIMIT OFFSET", async function() {
+      const limit = 1;
+      const offset = 2;
+
+      const querySpec = {
+        query: `SELECT * FROM root r OFFSET ${offset} LIMIT ${limit}`
+      };
+      const options = {
+        enableCrossPartitionQuery: true,
+        maxItemCount: 2
+      };
+
+      // validates the results size and order
+      await executeQueryAndValidateResults({
+        query: querySpec,
+        options,
+        expectedCount: 1
+      });
+    });
+
+    it("Validate filtered LIMIT OFFSET", async function() {
+      const limit = 1;
+      const offset = 2;
+
+      // an order by query with explicit ascending ordering
+      const querySpec = {
+        query: `SELECT * FROM root r WHERE r.number > 5 OFFSET ${offset} LIMIT ${limit}`
+      };
+      const options = {
+        enableCrossPartitionQuery: true,
+        maxItemCount: 2
+      };
+
+      // validates the results size and order
+      await executeQueryAndValidateResults({
+        query: querySpec,
+        options,
+        expectedCount: 1
+      });
+    });
+
+    it("Validate OrderBy Query With ASC and LIMIT 2 and OFFSET 10", async function() {
+      const limit = 2;
+      const offset = 10;
+
+      // an order by query with explicit ascending ordering
+      const querySpec = {
+        query: `SELECT * FROM root r order by r.spam ASC OFFSET ${offset} LIMIT ${limit}`
+      };
+      const options = {
+        enableCrossPartitionQuery: true,
+        maxItemCount: 2
+      };
+
+      const expectedOrderedIds = documentDefinitions
+        .sort(compare("spam"))
+        .map(function(r) {
+          return r["id"];
+        })
+        .splice(offset, limit);
+
+      // validates the results size and order
+      await executeQueryAndValidateResults({
+        query: querySpec,
+        options,
+        expectedOrderIds: expectedOrderedIds
+      });
+    });
+
+    it("Validate OrderBy Query With ASC and LIMIT 0 and OFFSET 5", async function() {
+      const limit = 5;
```

Comment:
Since you can pro grammatically generate the expected value, can you use nested loops for different combinations of offset, limit, and maxItemCount?

Reply:
Prefer not to at the moment. The entire file needs work, but for now would like to keep it consistent with the other test.